arbnode/node.go

```diff
@@ -1477,6 +1474,9 @@ func (n *Node) StopAndWait() {
 	if n.TxStreamer.Started() {
 		n.TxStreamer.StopAndWait()
 	}
+	if n.BroadcastServer != nil && n.BroadcastServer.Started() {
```

Comment:
for future generations: please add a comment here explaining why this is after TxStreamer